shell/platform/windows/flutter_windows_engine_unittests.cc

```diff
@@ -639,13 +639,16 @@ TEST_F(FlutterWindowsEngineTest, TestExit) {
   FlutterWindowsEngineBuilder builder{GetContext()};
   builder.SetDartEntrypoint("exitTestExit");
   bool finished = false;
-  bool did_call = false;
 
-  std::unique_ptr<FlutterWindowsEngine> engine = builder.Build();
+  auto window_binding_handler =
+      std::make_unique<::testing::NiceMock<MockWindowBindingHandler>>();
+  MockFlutterWindowsView view(std::move(window_binding_handler));
+  view.SetEngine(builder.Build());
+  FlutterWindowsEngine* engine = view.GetEngine();
 
-  EngineModifier modifier(engine.get());
+  EngineModifier modifier(engine);
```

Comment:
Should we use `FlutterWindowsEngineBuilder` instead and introduce an API to set the lifecycle manager?

Reply:
Would this mean adding a lifecycle manager field to `FlutterDesktopEngineProperties`? I believe that struct currently just has string/int values. If not, how would you think to do this?

Reply:
I would still use the `EngineModifier` under the hood. Feel free to ignore this suggestion if this proves to be painful